PocketKit/Sources/PocketKit/Home/CompactHomeCoordinator.swift

```diff
@@ -48,40 +46,27 @@ class CompactHomeCoordinator: NSObject {
         navigationController.popToRootViewController(animated: false)
         isResetting = true
 
-        model.$selectedReadableType.receive(on: DispatchQueue.main).sink { [weak self] readableType in
-            switch readableType {
-            case .savedItem(let viewModel):
-                self?.show(viewModel)
-            case .recommendation(let viewModel):
-                self?.show(viewModel)
-            case .none:
-                self?.readerSubscriptions = []
-            }
+        model.$selectedReadableType.sink { [weak self] readableType in
+            self?.show(readableType)
         }.store(in: &subscriptions)
 
-        model.$selectedSlateDetailViewModel.receive(on: DispatchQueue.main).sink { [weak self] viewModel in
-            self?.show(viewModel)
+        model.$selectedRecommendationToReport.sink { [weak self] recommendation in
```

Comment:
why did we decide not to receive these on main?

Reply:
The main reason is because we have a property within these coordinators called `isResetting` that we use to determine if we should animate certain transitions.

If we're resetting –i.e. if we're transitioning size classes– then we want to skip animations. If we receive these messages on the main queue, then the `isResetting` property will already be set to `false` by the time these closures are executed. By removing the `receive(on: ...)`, the closures will be executed immediately so we can count on the `isResetting` property to be correct.

Reply:
The second reasons is that it isn't necessary to explicitly jump to the main queue because all of these properties are already being set on the main thread